### PR TITLE
u2: fix custom dlvf & ulvf. resolves #14457

### LINKED ISSUE #14457: [u2]: incorrect downloadvolumefactor
### Have you checked our Troubleshooting page for your issue?

- [X] I have checked the Troubleshooting page

### Is there already an issue for your problem?

- [X] I have checked older issues, open and closed

### Have you read our Contributing Guidelines?

- [X] I have read the Contributing Guidelines

### Environment

```markdown
Jackett Version v0.21.205
```

### Description

a torrent with 0.3x download and 3x upload is parsed as `<torznab:attr name="downloadvolumefactor" value="0" />` and `<torznab:attr name="uploadvolumefactor" value="3" />`

image

### Logged Error Messages

```text
null
```

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by ilike2burnthing**
#### Description
fix custom dlvf & ulvf for U2

#### Issues Fixed or Closed by this PR

* Fixes #14457
* Closes #14460

### CODE REVIEW

File: src/Jackett.Common/Definitions/u2.yml [hunk 1]
```diff
@@ -156,11 +156,21 @@ search:
       selector: td:nth-child(7)
     grabs:
       selector: td:nth-child(8)
+    downloadvolumefactor_custom:
+      selector: td.embedded > img.arrowdown + b
+      filters:
+        - name: regexp
+          args: (.*)x
```
Comment (trim21) [inline src/Jackett.Common/Definitions/u2.yml:163]:
```suggestion
          args: (.*)X
```

File: src/Jackett.Common/Definitions/u2.yml [hunk 2]
```diff
@@ -156,11 +156,21 @@ search:
...
+    uploadvolumefactor_custom:
+      selector: td.embedded > img.arrowup + b
+      filters:
+        - name: regexp
+          args: (.*)x
```
Comment (trim21) [inline src/Jackett.Common/Definitions/u2.yml:168]:
```suggestion
          args: (.*)X
```

**Comment by ilike2burnthing** [comment]
Working?

**Comment by ilike2burnthing** [comment]
Whoops, that was my mistake, should have been optional. See if it's working now.

**Comment by trim21** [comment]
`{{ .Result.uploadvolumefactor_custom }}` and `{{ .Result.downloadvolumefactor_custom }}` are the expected values(for example `2.00` and `0.30`), but still not working


`System.Exception: Error while parsing field=downloadvolumefactor, selector=, value={{ .Result.downloadvolumefactor_custom }}: Input string was not in a correct format.`

**Comment by trim21** [comment]
```text
System.Exception: Error while parsing field=downloadvolumefactor, selector=, value={{ .Result.downloadvolumefactor_custom }}: Input string was not in a correct format.
 ---> System.FormatException: Input string was not in a correct format.
   at System.Number.ThrowOverflowOrFormatException(ParsingStatus status, TypeCode type)
   at System.Double.Parse(String s, NumberStyles style, IFormatProvider provider)
   at Jackett.Common.Utils.ParseUtil.CoerceDouble(String str) in ./Jackett.Common/Utils/ParseUtil.cs:line 50
   at Jackett.Common.Indexers.CardigannIndexer.ParseFields(String value, String FieldName, ReleaseInfo release, List`1 FieldModifiers, Uri searchUrlUri) in ./Jackett.Common/Indexers/CardigannIndexer.cs:line 2210
   at Jackett.Common.Indexers.CardigannIndexer.PerformQuery(TorznabQuery query) in ./Jackett.Common/Indexers/CardigannIndexer.cs:line 1782
   --- End of inner exception stack trace ---
   at Jackett.Common.Indexers.CardigannIndexer.PerformQuery(TorznabQuery query) in ./Jackett.Common/Indexers/CardigannIndexer.cs:line 1792
```

**Comment by trim21** [comment]
Is this doable? Or just go back to hardcode all possible values

**Comment by ilike2burnthing** [comment]
Make the following change (adding `strdump`):

```yml
    downloadvolumefactor_custom:
      selector: td.embedded > img.arrowdown + b
      optional: true
      filters:
        - name: regexp
          args: (.*)X
        - strdump
    uploadvolumefactor_custom:
      selector: td.embedded > img.arrowup + b
      optional: true
      filters:
        - name: regexp
          args: (.*)X
        - strdump
```

Restart Jackett and provide an enhanced log of the issue - https://github.com/Jackett/Jackett/wiki/Troubleshooting#how-to-provide-a-jackett-enhanced-log-for-a-ticket

Either something else is being included, or it's a simple formatting issue that I'm missing, but that @garfield69 or @mynameisbogdan might be able to spot with a fresh pair of eyes.

**Comment by trim21** [comment]
I have printed these values in rss item description, and I can conform they are expected value (`0.00`, `0.30`, `2.30` or ...etc)

**Comment by trim21** [comment]
I'm gussing, template here is not rendered

**Comment by trim21** [comment]
Yes the variable is not rendered

I add `Console.WriteLine` in ParseUtil.CoerceDouble

**Comment by trim21** [comment]
nevermind, I'll fix this

**Comment by trim21** [comment]
this PR works is https://github.com/Jackett/Jackett/pull/14462/files is merged

**Comment by garfield69** [comment]
Nice, thank you.
As for the strdump, see https://github.com/Jackett/Jackett/wiki/Definition-format#strdump

File: src/Jackett.Common/Definitions/u2.yml [hunk 3]
```diff
@@ -156,11 +156,23 @@ search:
       selector: td:nth-child(7)
     grabs:
       selector: td:nth-child(8)
+    downloadvolumefactor_custom:
+      selector: td.embedded > img.arrowdown + b
+      optional: true
+      filters:
+        - name: regexp
+          args: (.*)X
```
[the remark this answers is not in the document]
Comment (garfield69) [reply src/Jackett.Common/Definitions/u2.yml:164]: No reason. But it is not really likely that this field will have anything other than 3X or 3.00X for example, so we don't really have to guard against stray info appearing in the field. 
But your suggestion does eliminate possible future contaminations, given this is a custom field in the first place ;-)

[the remark this answers is not in the document]
Comment (trim21) [reply src/Jackett.Common/Definitions/u2.yml:164]: don't worry, I'll come back to fix this if this regex is broken.

**Pull request merged**